## Problem 2

In [7]:
ls -lsh

total 2.4M
1.2M -rw-rw-r-- 1 osboxes osboxes 1.2M Sep  5 21:12  conv.txt
4.0K -rw-rw-r-- 1 osboxes osboxes  147 Sep  5 21:12  DISCLAIMER
4.0K drwxrwxr-x 3 osboxes osboxes 4.0K Sep  6 02:48  Exam_1_Answers
8.0K -rw-rw-r-- 1 osboxes osboxes 5.5K Sep  6 02:47 'Exam Week 1.ipynb'
4.0K -rw-rw-r-- 1 osboxes osboxes 3.9K Sep  5 21:12 'Exam Week 2.ipynb'
8.0K -rw-rw-r-- 1 osboxes osboxes 4.2K Sep  5 21:12  Germplasm.tsv
4.0K -rw-rw-r-- 1 osboxes osboxes  141 Sep  5 21:12  how_to_convert.txt
4.0K -rw-rw-r-- 1 osboxes osboxes 1.1K Sep  5 21:12  LICENSE
4.0K -rw-rw-r-- 1 osboxes osboxes  637 Sep  5 21:12  LocusGene.tsv
1.2M -rw-rw-r-- 1 osboxes osboxes 1.2M Sep  5 21:12  Locus_Germplasm_Phenotype_20130122.txt
4.0K -rw-rw-r-- 1 osboxes osboxes  743 Sep  5 21:12  README.md


The command ls lists everything inside the directory. -l flag shows them in a list (so we are able to see ownership), -s shows the size, and -h shows the results in a way is easy to read for humans.

About the ownership, we can see there is only one dierctory, which is the one starting by "d", and there are 10 files, startig by "-". Then,we can see every element from this directory is readable and writeble by the user (first r and first w), and only the directory is executable by the owner (first x in the directory), that can be noted by the absence of x in the 4th column, and the "-" we can find instead.

The second r, w and x refer to the group, and the third ones refer to everyone.

## Problem 3

In [9]:
cd ..
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


As the header line is the first line of the document, we can use the command head, and specify to only include the first line of the file we are interested in.

## Problem 4

In [11]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


For this task, we can use the command wc (word count), and to specify to count the lines and not the words with the -l flag.

## Problem 5

In [14]:
tail -7215 Locus_Germplasm_Phenotype_20130122.txt > Data_only.csv
wc -l Data_only.csv

7215 Data_only.csv


As we know from the last Problem that our file has 7216 lines, we can take the last 7215 and direct the output to the new file. Then we use wc again to check that it has the expected number of lines

## Problem 6

In [ ]:
grep -nai "\Wroot\W" Locus_Germplasm_Phenotype_20130122.txt

We use the command grep in order to find all places where the word root is. Then, we use the flags -a because we dont know the source, and the txt file could be comming from windows, so it is better to work in binary; and -n, because we want the lines; and -i, because it could be the first word of a sentence. Also, we use the word root surrounded by something that is not a word character, as it could be preceded or followed by a "." or a "-".

## Problem 7

In [107]:
grep -ai "\Wroot\W" Locus_Germplasm_Phenotype_20130122.txt | grep -E -oia '^at[1-5]g[0-9]+\s' > Root-associated-Loci.txt

We take the same command we use for Problem 6, but not including the n, as we will need the line to start by the AGI locus code and not by the name of the line. Then, we use the pipe to send the output of the first grep to another grep, which is going to pick only the locus code (starting by "a", then "t", then a number from 1 to 5, a "g", and then 5 numbers from 0 to 9 and a tab) and redirect it to the file specified in the Problem 7.

When doing this Problem, I assumed we are only looking for the Locus code that match the AGI, and not looking at the ones on the bottom, that doesn't match this format.

## Problem 8

In [ ]:
grep -ai "\Wroot\W" Locus_Germplasm_Phenotype_20130122.txt | grep -E -oia '\s[0-9]+$' > Root-associated-Publications.txt

We use the same strategy we used before, but in the second grep we find the last characters of a line that contains only numbers and are preceded by a space (a tab in this case).

## Problem 9

In [110]:
grep -E -aio "^at[1]" Root-associated-Loci.txt | head -1
grep -E -aio "^at[2]" Root-associated-Loci.txt | head -1
grep -E -aio "^at[3]" Root-associated-Loci.txt | head -1
grep -E -aio "^at[4]" Root-associated-Loci.txt | head -1
grep -E -aio "^at[5]" Root-associated-Loci.txt | head -1 

AT1
AT2
AT3
AT4
AT5


We will grep only the ones for the first chromosome, and only the part that specify which chromosome, then for the second chromosome, and so on. Then, we will display them using head -1, so we can clearly see 1 the first result for each chromosome, and only the first part, in order to not overwhelm the the screen.

## Problem 10

Seems I did correctly last exercise, so I am skipping this part (wohoo!).